gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BomConflictCheckTask.java

```diff
@@ -0,0 +1,155 @@
+                            .forEach(artifactName -> resolvedConflicts.put(artifactName, propName));
+
+                    recommendations.entrySet()
+                            .stream()
+                            .filter(entry -> entry.getKey().matches(regex))
+                            .forEach(entry -> {
+                                conflicts.add(new Conflict(propName, propVersion, entry.getKey(), entry.getValue()));
+                                resolvedConflicts.remove(entry.getKey());
+                            });
+                    return null;
+                });
+
+        //Critical conflicts are versions.props line that only override bom recommendations with same version
+        //so it should avoid considering the case where a wildcard also pin an artifact not present in the bom
+        List<Conflict> critical = conflicts.stream()
+                .filter(c -> c.getBomVersion().equals(c.getPropVersion()))
+                .filter(c -> !resolvedConflicts.containsValue(c.getPropName()))
+                .collect(Collectors.toList());
+
+        if (!conflicts.isEmpty()) {
+            System.out.println("There are conflicts between versions.props and the bom:");
+            System.out.println(conflictsToString(conflicts, resolvedConflicts));
+
+            if (!critical.isEmpty()) {
+                throw new RuntimeException("Critical conflicts between versions.props and the bom "
+                        + "(overriding with same version)\n" + conflictsToString(critical, resolvedConflicts));
+            }
+        }
+
+    }
+
+    private String conflictsToString(List<Conflict> conflicts, Map<String, String> resolvedConflicts) {
+        return conflicts.stream()
+                .map(conflict -> conflict.details(resolvedConflicts))
+                .collect(Collectors.joining("\n"));
+    }
+
+    private static class Conflict {
+        private String propName;
+        private String propVersion;
+        private String bomName;
+        private String bomVersion;
+
+        Conflict(String propName, String propVersion, String bomName, String bomVersion) {
+            this.propName = propName;
+            this.propVersion = propVersion;
+            this.bomName = bomName;
+            this.bomVersion = bomVersion;
+        }
+
+        public String getPropName() {
+            return propName;
+        }
+
+        public String getPropVersion() {
+            return propVersion;
+        }
+
+        public String getBomName() {
+            return bomName;
+        }
+
+        public String getBomVersion() {
+            return bomVersion;
+        }
+
+        public String criticalString(Map<String, String> resolvedConflicts) {
+            if (!getBomVersion().equals(getPropVersion())) {
+                return "non critical: prop version not equals to bom version. (remove if unnecessary override)";
+            } else if (resolvedConflicts.containsValue(getPropName())) {
```

Comment:
using contains value indicates something is broken

Reply:
you're using a hashmap as a set - let's either use a hashmap and not lookup over values or use a set

Reply:
I need to be able to remove values based on the keys